test/strscan/test_stringscanner.rb

```diff
@@ -946,6 +946,18 @@ def test_scan_integer_encoding
     end
   end
 
+  def test_scan_integer_matched
+    omit("not implemented on TruffleRuby") if RUBY_ENGINE == "truffleruby"
+
+    s = create_string_scanner("42abc")
+    assert_equal(42, s.scan_integer)
+    assert_equal("42", s.matched)
+
+    s = create_string_scanner("42abc")
+    assert_equal(273084, s.scan_integer(base: 16))
```

Comment:
How about using hex literal for easy to read?

```suggestion
    assert_equal(0x42abc, s.scan_integer(base: 16))
```